x/mongo/driver/topology/topology_test.go

```diff
@@ -230,6 +232,29 @@ func TestServerSelection(t *testing.T) {
 			t.Fatalf("did not receive error from server selection")
 		}
 	})
+	t.Run("Context Deadline Error", func(t *testing.T) {
+		desc := description.Topology{}
+		topo, err := New()
+		noerr(t, err)
+		subCh := make(chan description.Topology, 1)
+		subCh <- desc
+		resp := make(chan error)
+		timeout := make(chan time.Time)
+		callback := func() {
+			state := newServerSelectionState(selectContextError, timeout)
+			ctx, cancel := context.WithTimeout(context.Background(), 50*time.Millisecond)
+			defer cancel()
+			_, err := topo.selectServerFromSubscription(ctx, subCh, state)
+			resp <- err
+			return
+		}
+		go func() {
```

Comment:
Why is this being called in a goroutine? `assert.Soon` will return as soon as the provided callback returns or the timeout elapses, so you should be able to call this synchronously.